src/Blazored.Modal/BlazoredModal.razor.cs

```diff
@@ -1,113 +1,82 @@
-        public void SetTitle(string title)
-        {
-            Title = title;
-            StateHasChanged();
-        }
+        private readonly Collection<ModalReference> Modals = new Collection<ModalReference>();
+        private readonly ModalOptions GlobalModalOptions = new ModalOptions();
 
         protected override void OnInitialized()
         {
-            ((ModalService)ModalService).OnShow += ShowModal;
-            ((ModalService)ModalService).CloseModal += CloseModal;
+            ((ModalService)ModalService).OnModalInstanceAdded += Update;
+            NavigationManager.LocationChanged += CancelModals;
+
+            GlobalModalOptions.Class = Class;
+            GlobalModalOptions.DisableBackgroundCancel = DisableBackgroundCancel;
+            GlobalModalOptions.HideCloseButton = HideCloseButton;
+            GlobalModalOptions.HideHeader = HideHeader;
+            GlobalModalOptions.Position = Position;
         }
 
-        private async void ShowModal(string title, RenderFragment content, ModalParameters parameters, ModalOptions options)
+        private async void CancelModals(object sender, LocationChangedEventArgs e)
         {
-            Title = title;
-            Content = content;
-            Parameters = parameters;
-
-            SetModalOptions(options);
+            foreach (var modalReference in Modals)
+            {
+                modalReference.Dismiss(ModalResult.Cancel());
+            }
 
-            IsVisible = true;
+            Modals.Clear();
             await InvokeAsync(StateHasChanged);
         }
 
-        private async void CloseModal()
+        internal void CloseInstance(Guid Id)
         {
-            IsVisible = false;
-            Title = "";
-            Content = null;
-            ComponentStyle = "";
-
-            await InvokeAsync(StateHasChanged);
+            CloseInstance(Id, ModalResult.Ok<object>(null));
         }
 
-        private void HandleBackgroundClick()
+        internal void CloseInstance(Guid Id, ModalResult result)
         {
-            if (ComponentDisableBackgroundCancel) return;
+            var reference = Modals.SingleOrDefault(x => x.Id == Id);
 
-            ModalService.Cancel();
+            if (reference != null)
+            {
+                reference.Dismiss(result);
+                Modals.Remove(reference);
+                StateHasChanged();
+            }
         }
 
-        private void SetModalOptions(ModalOptions options)
+        internal void CancelInstance(Guid Id)
         {
-            ComponentHideHeader = HideHeader;
-            if (options.HideHeader.HasValue)
-                ComponentHideHeader = options.HideHeader.Value;
-            
-            ComponentHideCloseButton = HideCloseButton;
-            if (options.HideCloseButton.HasValue)
-                ComponentHideCloseButton = options.HideCloseButton.Value;
+            var reference = Modals.SingleOrDefault(x => x.Id == Id);

```

Comment:
Could you not just call CloseInstance in this function, like this:
```suggestion
            CloseInstance(Id, ModalResult.Cancel());
```
basically replace this function with this one line?